frontend/src/common/MultiFileInput.tsx

```diff
@@ -47,43 +49,39 @@ export default function MultiFileInput({
         const updatedFiles = newFiles.concat(
           files.filter((file) => !newFiles.some((newFile) => newFile.name === file.name)),
         )
-        setFiles(updatedFiles)
+        onChange(updatedFiles)
       } else {
-        setFiles(newFiles)
+        onChange(newFiles)
       }
     }
   }
 
   return (
-    <Box sx={{ width: fullWidth ? '100%' : 'unset' }}>
-      <label htmlFor={id}>
-        <Input
-          style={{ margin: '10px' }}
-          id={id}
-          type='file'
-          onInput={handleFileChange}
-          multiple
-          accept={accepts}
-          disabled={disabled}
+    <Box sx={{ ...(fullWidth && { width: '100%' }) }}>
+      {!readOnly && (
+        <Box mb={2}>
+          <label htmlFor={htmlId}>
+            <Button
+              component='span'
+              variant='outlined'
+              disabled={disabled}
+              sx={{ ...(fullWidth && { width: '100%' }) }}
```

Comment:
I think `<Button />` has a `fullWidth` property so we can probably do `<Button fullWidth={fullWidth} />`